bin/oats

```diff
@@ -2,11 +2,48 @@
 
 const program = require("commander")
 const { generate } = require("../dist/generate")
+const version = require("../package.json").version
+const fs = require("fs")
+const parts = ["types", "request", "operations"]
+const truethy = x => !["false","no","0"].includes(x)
 
-program.arguments("<openapi spec>").action(async spec => {
-  const output = await generate(spec)
+let openApiSpec = ""
+program
+  .version(version)
+  .description("An opinionated OpenAPI doc to TypeScript HTTP client generator")
+  .option(
+    "-i, --include <parts>",
+    "include only specified code parts",
+    function(val) {
+      program.types = program.request = program.operations = false
+      val.split(",").forEach(elem => {
+        if (elem){
+          if (!parts.includes(elem)) {
+            throw new Error(
+              `Unsupported part '${elem}', supported parts are: ${parts}`
+            )
+          }
+          program[elem] = true
+        }
+      })
+    },
+    "types,request,operations"
+  )
+  .option("-p, --prettier [true/false]", "prettier output code", truethy, true)
+  .option("--withDoc [true/false]", "document generated types", truethy, true)
```

Comment:
The `truethy` function is very confusing to reason about, since it negates the values accepted.

And at the end of the day, I don't think we need to get super clever here with the `truthy` function. It's fine only accepting `true` and `false`, since the tool says those are the values this argument accepts are `true` and `false.

Reply:
OK, will replace with a more verbose
``` js
const parseBoolFlag = x => {
  if (x === 'true'){
    return true
  }
  if (x === 'false'){
    return false
  }
  throw new Error("Unsupported flag value, true/false is required!")
}
```